client/verta/verta/pipeline/_pipeline_graph.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+
+from typing import Any, Dict, List
+
+from verta._internal_utils._utils import Configuration, Connection
+from ._pipeline_step import PipelineStep
+
+
+class PipelineGraph:
+    """
+    A collection of PipelineSteps to be run as a single inference pipeline.
+
+    Parameters
+    ----------
+    steps : list of :class:`~verta.pipeline.PipelineStep`
+        List of all possible steps of the pipeline. Ordering of steps in the pipeline
+        itself is determined by the predecessors provided to each step, thus ordering
+        of this list is irrelevant.
+
+    Attributes
+    ----------
+        steps: list of :class:`~verta.deployment.PipelineStep`
+            List of PipelineSteps comprising all possible steps in the PiplineGraph.
+    """
+
+    def __init__(self, steps: List[PipelineStep]):
+        self._steps = steps
+
+    def __repr__(self):
+        return "\n".join((f"\nPipelineGraph steps:\n{self._format_steps()}",))
+
+    def _format_steps(self):
+        """Format steps for improved readability in __repr__() function."""
+        return "\n".join([str(s) for s in self._steps])
```

Comment:
Since this is for `__repr__()`, it'd probably be more accurate to use

```suggestion
        return "\n".join([repr(s) for s in self._steps])
```